new --reload-extra-file option

garbas opened the pull request:
I quickly added an option which is implemented but not exposed in the command line.
Would something like this be excepted? If yes I'll add the documentation for the command line option.

/cc @tilgovi (author of ``gunicorn/reloader.py``)

tilgovi commented:
Seems okay to me!

tilgovi commented:
As far as I can tell, nothing will break if you don't validate that the files exist, but maybe it's the right idea so that users don't silently watch the wrong thing?

garbas commented:
@tilgovi I usually follow the "fail early" rule and if somebody provides an option that points to a file which we should watch and that file does not exists it is very likely to be a typo. I would assume 99% of cases is like this. In the other 1%, you can always use ``reloader.py`` module programmaticaly and skip the validation.

I added the description for new option and squashed the work in one commit. As far as I can tell this looks ready to be merged, if not let me know wha

tilgovi commented on gunicorn/config.py:
```diff
@@ -379,6 +388,17 @@ def validate_list_string(val):
 
 
+def validate_list_of_existing_files(val):
+    if not val:
```
Maybe remove these few lines and replace with a call to `validate_list_string`?

tilgovi commented on gunicorn/config.py:
```diff
@@ -368,6 +368,15 @@ def validate_string(val):
 
+def validate_file_exists(val):
+    val = validate_string(val)
+    if val is None:
```
I'm not sure we want this here. For the string options, I think it's there for when the default is `None` if the option wasn't provided at all. In the case of this option, the default is an empty list.

tilgovi commented on gunicorn/config.py:
```diff
@@ -857,6 +877,20 @@ class ReloadEngine(Setting):
+    default = []
+    desc = """\
+        Extends --reload option to also watch and reload on additional files
+        (eg. templates, configurations, specifications, ...).
```
nit: "(e.g., templates, configurations, specifications, etc.)."

tilgovi commented:
Thanks, @garbas. Merged. 👍
